tensorflow/core/ops/array_ops.cc

```diff
@@ -2292,6 +2294,23 @@ Status SliceHelper(InferenceContext* c, ShapeHandle begin_value,
   return Status::OK();
 }
 
+// This SliceHelper processes the output shape of the `slice`
+// when the tensor of `sizes` is not available (null).
+// In this situation, we could still obtain partial information
+// from `MakeShapeFromShapeTensor` as a best effort operation.
+Status SliceHelper(InferenceContext* c, ShapeHandle begin_value,
+                   ShapeHandle sizes_value,
+                   std::vector<DimensionHandle>* dims) {
+  for (int i = 0; i < c->Rank(sizes_value); ++i) {
+    DimensionHandle dim = c->Dim(sizes_value, i);
+    DimensionHandle begin = c->Dim(begin_value, i);
+    DimensionHandle result = c->UnknownDim();
```

Comment:
you can also remove = c->UnknownDim(), as Subtract will set that.

Reply:
Done. And thanks a lot for the help!